### ISSUE TITLE
[v4] PUT /posts/{post_id}/patch contains misleading "file_ids" payload information

### ISSUE THREAD

**Issue by JoshStark** [mattermost/mattermost-api-reference#475]
The documentation relating to the PUT postPatch operation suggests that it is possible to update the array of associated `file_ids` for a single post. However this appears to conflict with the server code which explicitly states (in a code comment) that updating this field is not allowed and is thus ignored.

Would it be possible to get some clarification on this? Ideally why it is being ignored and whether or not the documentation is out of date. The specific code commit suggest this is part of a wider change around updating the EditAt field on updated posts.

**API Doc**: https://api.mattermost.com/#tag/posts%2Fpaths%2F~1posts~1%7Bpost_id%7D~1patch%2Fput

**Code Commit**: https://github.com/mattermost/mattermost-server/commit/7c9837d9b165792993163131522ee65c3f42d9ec

Many thanks.

**Comment by JoshStark** [comment]
Is it possible to get a follow-up for this please? The removed functionality is something I would have liked to use in one of my clients to the API. If it's better for me to move this Issue over to the mattermost-server GitHub repository, let me know.

Many thanks.

**Comment by DSchalla** [comment]
@JoshStark Sorry on the long delay for the response here. We only allow to set the file IDs of posts during their creation, since that is the only time an end user can upload attachments and add them to a post. We had in the post some problems with potential phishing vectors there as well.

Can you speak a little bit about your use case and what you'd like to accomplish? Happy to chat about potential solutions.

I'll update the documentation, thank you!

**Comment by JoshStark** [comment]
@DSchalla Thanks for getting back to me. I saw in the original change that the issue was around the post not displaying as "edited" if file_ids were changed. Was that the primary concern with the phishing vector? I assume this is due to it not being obvious that the file_ids had changed?

My particular use case is I have an "announcement" pinned post in a particular channel which has a set of uploaded documents which are useful/pertinent to that channel. The idea is that members of the channel may refer to these documents/images as and when they need to. However, these files are liable to change over time, so I would need to re-upload them whenever this occurs. Such posts also include a lot of descriptive text which provides context. Updating the text is easy, but to alter the uploaded images/documents I would need to delete the original post and recreate it, losing any link to the original post ID which some users may have permalinked/bookmarked elsewhere.

Hopefully this explains it well.
Thanks!
